harness/machine_types/linux_utilities.py

```diff
@@ -396,7 +396,7 @@ def build_executable(a_machine, new_env):
         #print("Setting env var", eu, "=", v)
         os.putenv(eu, v)
         message += f"Set build environment variable {eu}={v}\n"
```

Comment:
This also has different outcomes if new_env={} (empty dict).

Reply:
What follows is a for loop on new_env, so if an empty dict was passed in the for loop wouldn't execute any way. There is nothing else to be executed so effectively seems the same. That might change if we wanted to add additional code outside of the for loop to execute even with an empty dict.

The sole purpose of this line seems to be to protect the for loop from a None, which would break. I lean towards leaving it as-is but not sure it makes much difference.